lib/functions_framework.rb

```diff
@@ -142,6 +143,30 @@ def http name = DEFAULT_TARGET, &block
       self
     end
 
+    ##
+    # Define a function that responds to Typed invocations over HTTP.
+    #
+    # You must provide a name for the function, and a block that implements the
+    # function. The block should take a single `Hash` argument which will be the
+    # JSON decoded request payload. It should return a `Hash` response which
+    # will be JSON encoded and written to the response.
+    #
+    # ## Example
+    #     FunctionsFramework.http "my-sum-function" do |add_request|
+    #       {sum: add_request["num1"] + add_response["num2"]}
+    #     end
+    #
+    # @param name [String] The function name. Defaults to {DEFAULT_TARGET}
+    # @param request_type An optional class which will be used to decode the
```

Comment:
We need a type for `request_type`. We could use `[Class]` but I think better would be `[#decode_json]` since that's the actual interface we need the "type" object to implement.